### ISSUE TITLE
jax.nn.silu gradient returns zero for large negative float64 input

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#39798]
### Description

`jax.nn.silu` appears to return an incorrect zero gradient for a large negative finite `float64` input.

For SiLU / Swish,

```
silu(x) = x * sigmoid(x)
```

Its derivative is:

```
silu'(x) = sigmoid(x) + x * sigmoid(x) * (1 - sigmoid(x))
```

At `x = -709.0`, the true derivative is approximately `-8.614807714413836e-306`, which is finite and representable in `float64`. However, JAX returns `0.0`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

x = -709.0
expected = -8.614807714413836e-306

out = float(jax.grad(lambda z: jax.nn.silu(z))(jnp.asarray(x, dtype=jnp.float64)))

print("jax:", out)
print("expected:", expected)

if not (math.isfinite(out) and abs(out - expected) <= 1e-6 * abs(expected)):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: 0.0
expected: -8.614807714413836e-306
```

### Expected result

```
-8.614807714413836e-306
```

The expected derivative is finite and representable in `float64`. JAX instead returns `0.0`, suggesting that the backward pass flushes or loses the tiny but nonzero derivative for large negative inputs.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

**Comment by jakevdp** [comment]
#38651

**Comment by zhiwenf** [comment]
It seems to be due to subnormal flushing. The flushing happens when JAX computes
c = logistic(a=-709.0) ≈ 1.2168e-308 -> 0  (**flushed to zero**) 
This value is representable in float64, but only as a subnormal number, since the smallest positive normal float64 value is approximately 2.2251e-308.

```
x = -709.0
expected = -8.614807714413836e-306

d = jax.grad(lambda z: jax.nn.silu(z))

print(jax.make_jaxpr(d)(jnp.asarray(x, dtype=jnp.float64)))
out = float(d(jnp.asarray(x, dtype=jnp.float64)))
```
The generated JAXPR is:

```
{ lambda ; a:f64[]. let
    _:f64[] b:f64[] c:f64[] = jit[
      name=silu
      jaxpr={ lambda ; a:f64[]. let
          c:f64[] = logistic a
          d:f64[] = sub 1.0:f64[] c
          b:f64[] = mul c d
          _:f64[] = mul a c
        in (_, b, c) }
    ] a
    e:f64[] = jit[
      name=silu
      jaxpr={ lambda ; b:f64[] c:f64[] a:f64[] f:f64[]. let
          g:f64[] = mul a f
          h:f64[] = mul f c
          i:f64[] = mul g b
          e:f64[] = add_any h i
        in (e,) }
    ] b c a 1.0:f64[]
  in (e,) }
```

The autodiff expression itself appears correct. The issue seems to be that the intermediate c = logistic(-709.0) is flushed to zero.

**Comment by lihe07** [comment]
The behavior is caused by XLA DAZ on CPU backends, and on GPU backends x=-709 returns −8.61481e-306 correctly. 

I believe this is intended, see https://github.com/jax-ml/jax/issues/37766#issuecomment-4482876732
